docs-src/design-decisions.md

```diff
@@ -0,0 +1,42 @@
+# Design Decisions
+
+Throughout the design of these APIs some significant design decisions have been
+made. This provides some background on the factors considered when making these
+decisions.
+
+## GatewayClass Scope
+
+### Problem
+There were a number of compelling reasons to make GatewayClass a
+namespace-scoped resource. 
+
+### Advantages of a Namespace-Scoped GatewayClass
+* Less potential for naming conflicts among different implementations.
+* Better modeling for implementations with multiple GatewayClass resources
```

Comment:
Hi @robscott,

Another way to state this, and perhaps it is a separate item for the list, is that this change enables using Service API on a management cluster designed to provision and manage other clusters and infrastructure needed by other clusters.

Reply:
👍Good catch, thanks! Will update that.

Reply:
I don't think that "management cluster" is a well-defined concept that has consistent meaning or semantics, so I kind of object to it as a Noun.  I'd rather spell out what that means.